sql_generators/glean_usage/events_stream.py

```diff
@@ -0,0 +1,130 @@
+"""Generate events stream queries for Glean apps."""
+
+import os
+import re
+from collections import namedtuple
+from pathlib import Path
+
+from bigquery_etl.config import ConfigLoader
+from sql_generators.glean_usage.common import (
+    GleanTable,
+    get_table_dir,
+    render,
+    table_names_from_baseline,
+    write_sql,
+)
+
+TARGET_TABLE_ID = "events_stream_live_v1"
+PREFIX = "events_stream"
+PATH = Path(os.path.dirname(__file__))
+
+
+class EventsStreamTable(GleanTable):
+    """Represents generated events_stream table."""
+
+    def __init__(self):
+        """Initialize events_stream table."""
+        GleanTable.__init__(self)
+        self.target_table_id = TARGET_TABLE_ID
+        self.prefix = PREFIX
+        self.no_init = False
+        self.per_app_enabled = True
+        self.per_app_id_enabled = True
+        self.across_apps_enabled = False
+        self.cross_channel_template = "cross_channel_events_stream.view.sql"
+        self.base_table_name = "events_v1"
+
+    def generate_per_app_id(
+        self,
+        project_id,
+        baseline_table,
+        output_dir=None,
+        use_cloud_function=True,
+        app_info=[],
+    ):
+        # Get the app ID from the baseline_table name.
+        # This is what `common.py` also does.
+        app_id = re.sub(r"_stable\..+", "", baseline_table)
+        app_id = ".".join(app_id.split(".")[1:])
+
+        # Skip any not-allowed app.
+        if app_id not in ConfigLoader.get(
+            "generate", "glean_usage", "event_stream", "app_ids", fallback=[]
+        ):
+            return
+
```

Comment:
Custom logic for generating the files. Could maybe be simplified, but by default `generate_per_app_id` expects a `query.sql` file which doesn't exist here